packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -42,34 +44,77 @@ function genConfig(compatVersion: number): CompatConfig[] {
         }];
     }
 
-    return [
+    const allOld = {
+        loader: compatVersion,
+        driver: compatVersion,
+        containerRuntime: compatVersion,
+        dataRuntime: compatVersion,
+    };
+
+    const configs: CompatConfig[] = [
         {
-            name: `compat N${compatVersion} - old loader + new runtime`,
+            name: `compat N${compatVersion} - old loader`,
             kind: CompatKind.Loader,
             compatVersion,
             loader: compatVersion,
+            // back-compat: remove driver in future releases
+            driver: compatVersion,
+        },
+        {
+            name: `compat N${compatVersion} - new loader`,
+            kind: CompatKind.Loader,
+            compatVersion,
+            ...allOld,
+            loader: undefined,
+            // back-compat: remove driver in future releases
+            driver: undefined,
         },
         {
-            name: `compat N${compatVersion} - new loader + old runtime`,
+            name: `compat N${compatVersion} - old container runtime`,
             kind: CompatKind.ContainerRuntime,
             compatVersion,
             containerRuntime: compatVersion,
-            dataRuntime: compatVersion,
         },
         {
-            name: `compat N${compatVersion} - new container runtime + old data runtime`,
+            name: `compat N${compatVersion} - new container runtime`,
+            kind: CompatKind.ContainerRuntime,
+            compatVersion,
+            ...allOld,
+            containerRuntime: undefined,
+        },
+        {
+            name: `compat N${compatVersion} - old data runtime`,
             kind: CompatKind.DataRuntime,
             compatVersion,
             dataRuntime: compatVersion,
         },
         {
-            name: `compat N${compatVersion} - old container runtime + new data runtime`,
+            name: `compat N${compatVersion} - new data runtime`,
             kind: CompatKind.LoaderAndContainerRuntime,
             compatVersion,
-            loader: compatVersion,
-            containerRuntime: compatVersion,
+            ...allOld,
+            dataRuntime: undefined,
         },
     ];
+
+    // back-compat: in future, make this unconditional
+    if (compatVersion === -1) {
```

Comment:
Long term, we might want to have a better way to declare the min version that a test should be enable for.